src/Serokell/Util/Version.hs

```diff
@@ -0,0 +1,43 @@
+module Serokell.Util.Version
+       ( -- * Git revision
+         retrieveGitRev
+       ) where
+
+import Universum
+
+import Instances.TH.Lift ()
+import System.Environment (lookupEnv)
+import System.Exit (ExitCode (..))
+import System.Process (readProcessWithExitCode)
+
+import qualified Data.Text as Text
+import qualified Language.Haskell.TH as TH
+import qualified Language.Haskell.TH.Syntax as TH
+
+{- | Gets the git revision.
+For example, you can use it configuring CLI options:
+
+@
+versionOption = infoOption
+    ("Git revision: " <> toString $(retrieveGitRev))
+    (long "version" <> help "Show version")
+@
+
+Note, that you'll need @-XTemplateHaskell@.
+-}
+retrieveGitRev :: TH.Q TH.Exp
+retrieveGitRev = do
+    cti <- TH.runIO $ Text.strip . fromString <$> retrieveGit
+    TH.lift cti
+  where
+    retrieveGit :: IO String
+    retrieveGit =
+        lookupEnv "GITREV" >>= maybe retrieveFromGitExecutable pure
+
+    retrieveFromGitExecutable :: IO String
+    retrieveFromGitExecutable = do
+        (exitCode, output, _) <-
+            readProcessWithExitCode "git" ["rev-parse", "--verify", "HEAD"] ""
+        pure $ case exitCode of
+            ExitSuccess -> output
+            _           -> fail "Couldn't retrieve 'git' revision"
```

Comment:
Let's return `"<unknown>"` instead of failing. Just in case somebody wants to call this function not from git repository.

Reply:
No, this defaulting would be bad and fragile. Someone calling not from a git repo can define `GITREV="<unknown>"` in the environment.

Reply:
@int-index Okay, that sounds reasonable. I just tried to follow up on @yorickvP suggestions. But maybe I just misunderstood what he means.